aspnet/2-structured-data/Models/DatastoreBookStore.cs

```diff
@@ -132,112 +122,77 @@ public DatastoreBookStore(string projectId)
 
         // [START create]
         public void Create(Book book)
         {
-            var result = CommitMutation(new Mutation()
-            {
-                InsertAutoId = new Entity[] { book.ToEntity() }
-            });
-            book.Id = result.MutationResult.InsertAutoIdKeys.First().Path.First().Id.Value;
+            // P0 Calling .ToInsert() is very weird.
+            // Having two very different ways to insert, update, etc. depending on whether or not
+            // I'm in a transaction is annoying.  I want one *interface* to do it.
+            // How about a NullTransaction where all operations are immediately committed, and the
+            // final .Commit() is a no-op?  Or, make DatastoreFoo and Transaction implement the
+            // same interface?
+            CommitResponse response = _datastore.Commit(_projectId, Mode.NonTransactional, new[] { book.ToEntity().ToInsert() });
+            Key key = response.MutationResults[0].Key;
+            book.Id = key.Path.First().Id;
         }
         // [END create]
 
         public void Delete(long id)
         {
-            CommitMutation(new Mutation()
-            {
-                Delete = new Key[] { id.ToKey() }
-            });
+            // Pretty good.
+            _db.Delete(id.ToKey());
+            var trans = _db.BeginTransaction();
+            trans.Delete(id.ToKey());
+            trans.Commit();
         }
 
         // [START list]
         public BookList List(int pageSize, string nextPageToken)
         {
-            var query = new Query()
+            var query = new Query("Book")
             {
                 Limit = pageSize,
-                Kinds = new[] { new KindExpression() { Name = "Book" } },
             };
 
             if (!string.IsNullOrWhiteSpace(nextPageToken))
-                query.StartCursor = nextPageToken;
+                query.StartCursor = Google.Protobuf.ByteString.CopyFromUtf8(nextPageToken);
 
-            var datastoreRequest = _datastore.Datasets.RunQuery(
-                datasetId: _projectId,
-                body: new RunQueryRequest() { Query = query }
-            );
-
-            var response = datastoreRequest.Execute();
-            var results = response.Batch.EntityResults;
-            var books = results.Select(result => result.Entity.ToBook());
+            var results = _db.RunQuery(query);
+            var books = results.Select(result => result.ToBook());
 
             return new BookList()
             {
                 Books = books,
+                // More string vs proto byte string warnings below.  Why?
                 NextPageToken = books.Count() == pageSize
-                    && response.Batch.MoreResults == "MORE_RESULTS_AFTER_LIMIT"
-                    ? response.Batch.EndCursor : null,
+                    && response.Batch.MoreResults == QueryResultBatch.Types.MoreResultsType.MoreResultsAfterCursor
+                    ? response.Batch.EndCursor.ToStringUtf8() : null,

```

Comment:
I suspect you should be using `var page = results.AsPages().WithFixedPageSize(pageSize).FirstOrDefault();`

You can then use `page.NextPageToken` to get the base64 representation of the `EndCursor`.